### ISSUE TITLE
only using single sample from batch in finetuning example?

### ISSUE THREAD

**Issue by evanatyourservice** [google-deepmind/gemma#48]
```
  # Exclude the last step as it does not appear in the targets.
  logits = logits[0, :-1]

  # Similarly, the first token cannot be predicteds.
  target_tokens = input_tokens[0, 1:]
  target_mask = input_mask[0, 1:]
```

The zero indexes there, are they only taking one example from the batch, should they not be `:`?

**Comment by Gopi-Uppari** [comment]
Hi @evanatyourservice,

Indexing with 0 only processes the first sample and ignores the rest of the batch, while using : processes all the samples in the batch.

Below code ensures that the model processes every example in the batch:

`logits = logits[:, :-1]`
`target_tokens = input_tokens[:, 1:]`
`target_mask = input_mask[:, 1:] `

Thank you.

**Comment by evanatyourservice** [comment]
Let me know if I can help by submitting a pr

**Comment by Gopi-Uppari** [comment]
Hi @evanatyourservice,

Please feel free to go ahead and submit the PR.

Thank you.

**Comment by Gopi-Uppari** [comment]
Hi @evanatyourservice,

Have you raised a PR ?

Could you please confirm if this issue is resolved for you with the above comment ? Please feel free to close the issue if it is resolved ?

Thank you.

**Comment by evanatyourservice** [comment]
Hi @Gopi-Uppari! Sorry I completely forgot about this, let me do a PR real quick right now

**Comment by evanatyourservice** [comment]
@Gopi-Uppari oh looks like someone else has gotten to it, all looks good! Have a nice day!
